cmu_graphics/cmu_graphics.py

```diff
@@ -208,8 +208,44 @@ def __iter__(self): return iter(self._shape)
     def __len__(self): return len(self._shape._shapes)
 
 class Sound(object):
+    number_of_sounds = 0
     def __init__(self, url):
-        self.sound = slNewSound(url)
+        if not pygame.mixer.get_init():
+            pygame.mixer.init()
+            pygame.mixer.set_num_channels(1)
+
+        if not isinstance(url, str):
+            callSpec = '{className}.{attr}'.format(className=t('Sound'), attr=t('url'))
+            err = t(
+                    '{{error}}: {{callSpec}} should be {{typeName}} (but {{value}} is of type {{valueType}})',
+                    {'error': t('TypeError'), 'callSpec': callSpec, 'typeName': 'string', 'value': repr(url), 'valueType': type(url).__name__}
+                    )
+            raise Exception(err)
+
+        Sound.number_of_sounds += 1
+        if pygame.mixer.get_num_channels()==Sound.number_of_sounds:
+            pygame.mixer.set_num_channels(Sound.number_of_sounds * 2)
+
+        if url.startswith('file://'):
+            url = url.split('//')[-1]
+        if url.startswith('http'):
+            for i in range(10):
+                try:
+                    response = webrequest.get(url)
+                    self.sound = io.BytesIO(response.read())
+                except:
+                    if i<9:
+                        continue
+                    else:
+                        raise Exception('Failed to load sound data')
+                break
```

Comment:
Request change: I don't think this is a good way to handle failed requests; something like [the image fetching logic](https://github.com/cmu-cs-academy/desktop-cmu-graphics/blob/ac23892e968a5b74abc669d5f2d3d405d56d1ee7/cmu_graphics/shape_logic.py#L440-L444) might be more appropriate here (also, `pyThrow` might be preferable over generic Python error types).

Reply:
The rationale behind this code is to make the cmu_graphics more robust; if a network error happens, we don't just want to directly crash the program and instead try again a couple of times. In this spirit, I actually think that the sound fetching logic should be updated to reflect the behaviour seen here

pyThrow isn't defined in cmu_graphics.py nor was it ever used there; instead, normal exception was used. pyThrow is only defined inside shape_logic.py and is merely a simple wrapper of the exception class.  We could do something about this, but if we are going to, we need to be consistent with it. So, any new thoughts on this?